AppCenter/AppCenterTests/MSChannelUnitDefaultTests.m

```diff
@@ -72,17 +67,12 @@ - (void)setUp {
    * dispatch_get_main_queue isn't good option for logsDispatchQueue because
    * we can't clear pending actions from it after the test. It can cause usages of stopped mocks.
    */
-  self.logsDispatchQueue = dispatch_queue_create("com.microsoft.appcenter.ChannelGroupQueue", DISPATCH_QUEUE_SERIAL);
```

Comment:
`setUp` is called before each test, so the only change is to drop the id. Please revert changes in all tests just because of this

Reply:
The queue ID only displays additional information in the debugger, and it does not affect the test process: https://developer.apple.com/documentation/dispatch/1453030-dispatch_queue_create
Before that, each test depended on the global variable dispatchQueueLogs. if we try to overwrite the value of the queue that started its execution, we run the risk of getting a critical error. Therefore, we wrote initialization of our queue on each test (we caught this situation when debugging).

Reply:
> we run the risk of getting a critical error
Can you explain?
Is this an intention of dealloc?
I couldn't find any changes that fixes the issue. Please explain.

Reply:
I don't think I understood why we don't create the queue in the setup anymore.

Reply:
All of our tests were dependent on the global queue variable. There are situations when the test was completed, but the queue still continues its execution. In the next test, the startup method changes the value of the current queue. Old queue continues its execution on the sut object, and causes conflicts.

Reply:
In the deadlock test, I changed the recursive mutex to semaphore. Recursive mutex did not wait for the background queue to start executing.

Reply:
Why does old queue continues its execution on the sut object isn't the old queue suspended? In this case are you referring to sut as in the sut of the new test or the sut of the old test?